php-checks/src/main/java/org/sonar/php/checks/phpunit/AssertTrueInsteadOfDedicatedAssertCheck.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks.phpunit;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.Map;
+import java.util.Optional;
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.PhpUnitCheck;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.expression.BinaryExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.tree.expression.UnaryExpressionTree;
+
+import static org.sonar.php.checks.utils.CheckUtils.skipParenthesis;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.NULL_LITERAL;
+
+@Rule(key = "S5785")
+public class AssertTrueInsteadOfDedicatedAssertCheck extends PhpUnitCheck {
+
+  private static final String MESSAGE = "Use %s instead.";
+  private static final String SECONDARY_MESSAGE = "%s is performed here, which is better expressed with %s.";
+  private static final ImmutableSet<String> ASSERT_METHOD_NAMES = ImmutableSet.of("assertTrue", "assertFalse");
+
+  private static final Map<ReplacementAssertion, ReplacementAssertion> COMPLEMENTS = ImmutableMap.<ReplacementAssertion, ReplacementAssertion>builder()
+    .put(ReplacementAssertion.NULL, ReplacementAssertion.NOT_NULL)
+    .put(ReplacementAssertion.NOT_NULL, ReplacementAssertion.NULL)
+    .put(ReplacementAssertion.SAME, ReplacementAssertion.NOT_SAME)
+    .put(ReplacementAssertion.NOT_SAME, ReplacementAssertion.SAME)
+    .put(ReplacementAssertion.EQUALS, ReplacementAssertion.NOT_EQUALS)
+    .put(ReplacementAssertion.NOT_EQUALS, ReplacementAssertion.EQUALS)
+    .build();
+
+  private enum ReplacementAssertion {
+    NULL("Null", "A null-check"),
+    NOT_NULL("NotNull", "A null-check"),
+    SAME("Same", "An object reference comparison"),
+    NOT_SAME("NotSame", "An object reference comparison"),
```

Comment:
Suggestion for the descriptions as it might not only concern objects: "A type-safe equality check"